UT Spatial Structure 6

AlexBrad1eyCT opened the pull request:
## Unit test

This scenario establishes a spatial structure for a shipyard. The intention is to show and test how spatial structures can be used to organize a project. The model utilizes new vertical and regional breakdown objects to organize the project and provide a skeleton for local placement.

The IFC files anre encoded RC3 so will need to be checked once checking tool is ready.
## Covered topics

*tick one/many*

- [x] project setup
- [ ] georeferencing
- [ ] alignment semantics
- [ ] alignment geometry
- [ ] linear placement
- [x] project breakdown structure / spatial structure
- [ ] terrain
- [ ] triangulated irregular network
- [ ] open cross profile
- [ ] sectioned surface & solid
- [ ] pavement & course
- [ ] surface feature
- [ ] earthworks
- [ ] geotechnics
- [ ] longitudinal products (barriers, guardrail, ...)
- [ ] interchange
- [ ] structural
- [ ] drainage
- [ ] signage
- [ ] road furniture
- [ ] rail furniture
- [ ] ports & waterways furniture
- [ ] bridge furniture

AlexBrad1eyCT committed:
Initial Setup

AlexBrad1eyCT committed:
Initial Setup

AlexBrad1eyCT committed:
readme draft

AlexBrad1eyCT committed:
readme draft

AlexBrad1eyCT committed:
First Draft of SpatialStructure-6

First unit test draft with ifc files, pictures & readme

AlexBrad1eyCT committed:
First Draft of SpatialStructure-6

First unit test draft with ifc files, pictures & readme

AlexBrad1eyCT committed:
readme corrections

AlexBrad1eyCT committed:
readme corrections

jmirtsch commented:
I'd suggest the transport elements should be contained in the space, not aggregating as per this image.

![210309 element containment](https://user-images.githubusercontent.com/791757/110441120-83291e00-810d-11eb-830b-dea47d626c94.png)

AlexBrad1eyCT committed:
Hierarchy error and temporary RC2 versions

corrected error where products were added as aggregations instead of containedInSpatialStructure

AlexBrad1eyCT committed:
Hierarchy error and temporary RC2 versions

corrected error where products were added as aggregations instead of containedInSpatialStructure

AlexBrad1eyCT commented:
@jmirtsch thanks for catching that the inevitable overlook of code reuse! Fixed in latest commit

AlexBrad1eyCT committed:
Validation corrections

AlexBrad1eyCT committed:
Validation corrections

AlexBrad1eyCT committed:
Merge remote-tracking branch 'origin/UT-SpatialStructure-6' into UT-SpatialStructure-6

pjanck commented:
Can you please merge the `main` branch again and rerun the tests? The checker just got updated.

AlexBrad1eyCT committed:
Merge remote-tracking branch 'bSI-InfraRoom/main' into UT-SpatialStructure-6

AlexBrad1eyCT committed:
corrections

AlexBrad1eyCT committed:
last uprating

pjanck commented:
How is error-fixing progressing? Having any difficulties?

pjanck commented on SpatialStructure-6/readme.md:
```diff
@@ -0,0 +1,134 @@
+			- `PredefinedType=IfcMarineFacilityTypeEnum.DRYDOCK`
+
+
+The application should be able to display a project hierarchy as shown in Figure 2.
+
+![Project Tree][figure 3]
```
Do you mean figure 3 here?

AlexBrad1eyCT replied:
Yep good spot, correction in next commit

AlexBrad1eyCT commented:
I'm sorting out some style issues in the revit export im extracting elements from, revit is exporting `IfcPresentationStyleAssignment` objects which are deprecated(causing most of the errors I think). And I couldn't find the delete for GGIfc (jon has told me now). This should sort that out. Will push when I update my cleanup method on my scripts.

jmirtsch commented:
> 
> 
> I'm sorting out some style issues in the revit export im extracting elements from, revit is exporting `IfcPresentationStyleAssignment` objects which are deprecated(causing most of the errors I think). And I couldn't find the delete for GGIfc (jon has told me now). This should sort that out. Will push when I update my cleanup method on my scripts.

I'll email you, I'll help make the toolkit upgrade the presentationstyleassignment rather than delete it.

AlexBrad1eyCT committed:
First Try Cleanup Coding

AlexBrad1eyCT committed:
last bit of corrections and cleanup

AlexBrad1eyCT committed:
forgot to change schema type back

AlexBrad1eyCT committed:
fixed IfcApplication Duplication

AlexBrad1eyCT commented:
Solved all errors, that relate to file, only thing left is a variation of the KIT problem with Where rules and on IfcElementType subtypes

AlexBrad1eyCT committed:
Merge remote-tracking branch 'bSI-InfraRoom/main' into UT-SpatialStructure-6

AlexBrad1eyCT committed:
Merge remote-tracking branch 'bSI-InfraRoom/main' into UT-SpatialStructure-6

AlexBrad1eyCT commented:
@pjanck  Sucess! PDT where rules all functioning! i will now merge!
